packages/widgets/src/tabbar.ts

```diff
@@ -156,6 +157,13 @@ export class TabBar<T> extends Widget {
    */
   readonly renderer: TabBar.IRenderer<T>;
 
+  /**
+   * The document to use with the tab bar.
+   *
+   * The default is the global `document` instance.
+   */
+  document: Document | ShadowRoot;
```

Comment:
I think this should be readonly, and only settable in the constructor.

Reply:
Change made as suggested.